src/Workleap.OpenApi.MSBuild/OasdiffManager.cs

```diff
@@ -1,35 +1,71 @@
     private readonly IProcessWrapper _processWrapper;
-    private readonly string _toolDirectory;
+    private readonly string _openApiToolsDirectory;
+    private readonly string _oasdiffDirectory;
 
-    public OasdiffManager(IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    public OasdiffManager(ILoggerWrapper loggerWrapper, IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
     {
+        this._loggerWrapper = loggerWrapper;
         this._httpClientWrapper = httpClientWrapper;
         this._processWrapper = processWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._oasdiffDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
     }
-
+    
     public async Task InstallOasdiffAsync(CancellationToken cancellationToken)
     {
-        Directory.CreateDirectory(this._toolDirectory);
+        this._loggerWrapper.LogMessage("Starting Oasdiff installation.");
+            
+        Directory.CreateDirectory(this._oasdiffDirectory);
 
         var oasdiffFileName = GetOasdiffFileName();
         var url = $"https://github.com/Tufin/oasdiff/releases/download/v{OasdiffVersion}/{oasdiffFileName}";
 
-        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._toolDirectory, oasdiffFileName), cancellationToken);
+        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._oasdiffDirectory, oasdiffFileName), cancellationToken);
         await this.DecompressDownloadedFileAsync(oasdiffFileName, cancellationToken);
+            
+        this._loggerWrapper.LogMessage("Oasdiff installation completed.");
+    }
+
+    public async Task RunOasdiffAsync(IEnumerable<string> openApiSpecFiles, IEnumerable<string> generatedOpenApiSpecFiles, CancellationToken cancellationToken)
+    {
+        var generatedOpenApiSpecFilesList = generatedOpenApiSpecFiles.ToList();
+        var oasdiffExecutePath = Path.Combine(this._oasdiffDirectory, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "oasdiff.exe" : "oasdiff");
+
+        foreach (var specFile in openApiSpecFiles)
+        {
+            this._loggerWrapper.LogMessage($"Starting Oasdiff comparison with {specFile}.");
+            
+            var fileName = Path.GetFileName(specFile);
+            var baseSpecRelativePath = Path.GetRelativePath(this._openApiToolsDirectory, specFile);
+            var newSpecRelativePath = Path.GetRelativePath(this._openApiToolsDirectory, generatedOpenApiSpecFilesList.First(x => x.Contains(fileName)));
+            await this._processWrapper.RunProcessAsync(oasdiffExecutePath, new[] { "diff", baseSpecRelativePath, newSpecRelativePath, "--exclude-elements", "description,examples,title,summary", "-f", "text" }, cancellationToken);
```

Comment:
Could be more explicit about the intent to put the arguments in variable that explains what we want:

var excludeElements = "--exclude descriptions,..."
var formatInText = "-f text"

Reply:
Following ScaffoldIt convention. Wouldn't the flags themselves provide more explanation (i.e. `--exclude-elements` variable name will be `excludeElements`. We are not providing any additional information here)

thoughts?

Reply:
It would make it explicit what is grouped together. Could also add info about the intent "excludeUnecessaryInformation,...."

I find it clearer to not have "magic string" used directly in parameters but wrap in var.

Overall it's nitpick